amida-tech/indaba-client
# Task/inba 436 review conditions

#### What's this PR do?
Conditionally displays the review panel for the surveys based on the stage. 

Regardless of all conditions, the review panel displays if the current stage is for "Review and Comment" or "Review and Edit." The review panel should NOT appear for "Complete Survey" or "Review."

The question pane should be disabled and unclickable if:
- The survey status is "draft."
- The task's start date is in the future.
- The stage is "Review."
- The stage is "Review and Comment."
- If the user viewing it is not the user assigned to it.

The review pane should be disabled if:
- The survey status is "draft."
- The task's start date is in the future.
- If the user viewing it is not the user assigned to it.

#### Related JIRA tickets:
#### How should this be manually tested?
Create a project. Create 4 stages, one for each of the stage types (complete survey, review, review and comment, review and edit). Make certain there is a survey attach to this with answers (easiest way to do this is to go into the `Products` table and put 1 for `surveyId` for the Product-- This loads basic questions).

Log out and log in as the user in question. Go through each task and check whether the review pane appears, and whether the questions and review pane are enabled based on the survey's status, and if the task has started yet via the start date. The easiest way to change the start date is in the `Tasks` table under `startDate`.

Finally, try logging in as another user and viewing the task. It should be disabled if you are not the user assigned to it. 
#### Any background context you want to provide?
#### Screenshots (if appropriate):
